Doesn't work with sbt 1.3.0-RC2?

According to @SethTisue, sbt-whitesource doesn't work with sbt 1.3.0-RC2 when used for some repos, in this case lightbend/scala-fortify-plugin.


 ```
sbt:scala-fortify> whitesourceUpdate
[info] Service URL is https://saas.whitesourcesoftware.com/agent
[info] Initiated WhiteSource Service
[info] Processing com.lightbend:scala-fortify:1.0.15
[error] java.lang.RuntimeException: Unexpected error
[error] at sbtwhitesource.BaseAction.execute(WhiteSource.scala:71)
[error] at sbtwhitesource.WhiteSourcePlugin$.$anonfun$projectSettings$11(WhiteSourcePlugin.scala:158)
[error] at sbtwhitesource.WhiteSourcePlugin$.$anonfun$projectSettings$11$adapted(WhiteSourcePlugin.scala:155)
[error] at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error] at sbt.std.Transform$$anon$4.work(Transform.scala:67)
[error] at sbt.Execute.$anonfun$submit$2(Execute.scala:280)
[error] at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:19)
[error] at sbt.Execute.work(Execute.scala:289)
[error] at sbt.Execute.$anonfun$submit$1(Execute.scala:280)
[error] at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:178)
[error] at sbt.CompletionService$$anon$2.call(CompletionService.scala:37)
[error] at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error] at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[error] at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error] at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[error] at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[error] at java.lang.Thread.run(Thread.java:748)
[error] Caused by: java.lang.RuntimeException: Multiple elements for
the same key (com.lightbend,scala-st-nodes):
[error] ModuleInfo(com.lightbend,scala-st-nodes,0.0.0+105-e8ef584e,Some((Artifact(scala-st-nodes,
jar, jar, Some(assembly), Vector(),
Some(file://///Users/tisue/.ivy2//local/com.lightbend/scala-st-nodes/0.0.0+105-e8ef584e/jars/scala-st-nodes-assembly.jar),
Map(artifactId -> scala-st-nodes, project.packaging -> jar,
project.version -> 0.0.0+105-e8ef584e, pom.groupId -> com.lightbend,
groupId -> com.lightbend, project.groupId -> com.lightbend, version ->
0.0.0+105-e8ef584e, pom.version -> 0.0.0+105-e8ef584e,
project.artifactId -> scala-st-nodes, pom.artifactId ->
scala-st-nodes),
None),/Users/tisue/.ivy2/local/com.lightbend/scala-st-nodes/0.0.0+105-e8ef584e/jars/scala-st-nodes-assembly.jar)))
[error] ModuleInfo(com.lightbend,scala-st-nodes,0.0.0+105-e8ef584e,Some((Artifact(scala-st-nodes,
src, jar, Some(sources), Vector(),
Some(file://///Users/tisue/.ivy2//local/com.lightbend/scala-st-nodes/0.0.0+105-e8ef584e/srcs/scala-st-nodes-sources.jar),
Map(artifactId -> scala-st-nodes, project.packaging -> jar,
project.version -> 0.0.0+105-e8ef584e, pom.groupId -> com.lightbend,
groupId -> com.lightbend, project.groupId -> com.lightbend, version ->
0.0.0+105-e8ef584e, pom.version -> 0.0.0+105-e8ef584e,
project.artifactId -> scala-st-nodes, pom.artifactId ->
scala-st-nodes),
None),/Users/tisue/.ivy2/local/com.lightbend/scala-st-nodes/0.0.0+105-e8ef584e/srcs/scala-st-nodes-sources.jar)))
[error] at scala.sys.package$.error(package.scala:30)
[error] at sbtwhitesource.package$KeyByAndMergeSyntax$.$anonfun$keyByAndMerge$2(package.scala:26)
[error] at scala.Option.getOrElse(Option.scala:138)
[error] at sbtwhitesource.package$KeyByAndMergeSyntax$.$anonfun$keyByAndMerge$1(package.scala:26)
[error] at sbtwhitesource.package$KeyByAndMergeSyntax$.$anonfun$keyByAndMerge$1$adapted(package.scala:20)
[error] at scala.collection.Iterator.foreach(Iterator.scala:941)
[error] at scala.collection.Iterator.foreach$(Iterator.scala:941)
[error] at scala.collection.AbstractIterator.foreach(Iterator.scala:1429)
[error] at scala.collection.IterableLike.foreach(IterableLike.scala:74)
[error] at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
[error] at scala.collection.AbstractIterable.foreach(Iterable.scala:56)
[error] at sbtwhitesource.package$KeyByAndMergeSyntax$.keyByAndMerge$extension(package.scala:20)
[error] at sbtwhitesource.BaseAction.moduleInfoByGA$1(WhiteSource.scala:161)
[error] at sbtwhitesource.BaseAction.moduleInfos$1(WhiteSource.scala:165)
[error] at sbtwhitesource.BaseAction.collectDependencyStructure(WhiteSource.scala:172)
[error] at sbtwhitesource.BaseAction.processProject(WhiteSource.scala:146)
[error] at sbtwhitesource.BaseAction.$anonfun$extractProjectInfos$2(WhiteSource.scala:82)
[error] at scala.collection.Iterator$$anon$10.next(Iterator.scala:459)
[error] at scala.collection.Iterator.foreach(Iterator.scala:941)
[error] at scala.collection.Iterator.foreach$(Iterator.scala:941)
[error] at scala.collection.AbstractIterator.foreach(Iterator.scala:1429)
[error] at scala.collection.generic.Growable.$plus$plus$eq(Growable.scala:62)
[error] at scala.collection.generic.Growable.$plus$plus$eq$(Growable.scala:53)
[error] at scala.collection.immutable.VectorBuilder.$plus$plus$eq(Vector.scala:662)
[error] at scala.collection.immutable.VectorBuilder.$plus$plus$eq(Vector.scala:639)
[error] at scala.collection.TraversableOnce.to(TraversableOnce.scala:313)
[error] at scala.collection.TraversableOnce.to$(TraversableOnce.scala:311)
[error] at scala.collection.AbstractIterator.to(Iterator.scala:1429)
[error] at scala.collection.TraversableOnce.toVector(TraversableOnce.scala:309)
[error] at scala.collection.TraversableOnce.toVector$(TraversableOnce.scala:309)
[error] at scala.collection.AbstractIterator.toVector(Iterator.scala:1429)
[error] at sbtwhitesource.BaseAction.extractProjectInfos(WhiteSource.scala:82)
[error] at sbtwhitesource.UpdateAction.doExecute(WhiteSource.scala:391)
[error] at sbtwhitesource.BaseAction.execute(WhiteSource.scala:68)
[error] at sbtwhitesource.WhiteSourcePlugin$.$anonfun$projectSettings$11(WhiteSourcePlugin.scala:158)
[error] at sbtwhitesource.WhiteSourcePlugin$.$anonfun$projectSettings$11$adapted(WhiteSourcePlugin.scala:155)
[error] at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error] at sbt.std.Transform$$anon$4.work(Transform.scala:67)
[error] at sbt.Execute.$anonfun$submit$2(Execute.scala:280)
[error] at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:19)
[error] at sbt.Execute.work(Execute.scala:289)
[error] at sbt.Execute.$anonfun$submit$1(Execute.scala:280)
[error] at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:178)
[error] at sbt.CompletionService$$anon$2.call(CompletionService.scala:37)
[error] at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error] at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[error] at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error] at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[error] at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[error] at java.lang.Thread.run(Thread.java:748)
```

### notes

see https://github.com/lightbend/sbt-whitesource/blob/ac79a5479f6f6f93cedea230eabfcebe121ed779/src/main/scala/sbtwhitesource/WhiteSource.scala#L326-L333

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,5 +1,5 @@
 language: scala
-jdk: oraclejdk8
+jdk: openjdk8
 script: project/travis.sh
 
 # Build only master and version tags - http://stackoverflow.com/a/31882307/463761

--- a/src/main/scala/sbtwhitesource/WhiteSource.scala
+++ b/src/main/scala/sbtwhitesource/WhiteSource.scala
@@ -338,7 +338,9 @@ object BaseAction {
         case (Some(_), None)                  => None
         case (None, None)                     => Some(m1)
         case (Some((a1, f1)), Some((a2, f2))) =>
-          if (a1.classifier == Some("native") && a2.classifier == None)
+          if (a1.classifier == a2.classifier && a1.`type` == a2.`type`)
+            Some(m1)
+          else if (a1.classifier == Some("native") && a2.classifier == None)
             Some(m2)
           else if (a1.classifier == None && a2.classifier == Some("native"))
             Some(m1)

